common/build-helper/numpy.sh

```diff
@@ -0,0 +1,36 @@
+#
+# numpy - build-helper for packages that compile against python3-numpy
+#
+# This build-helper makes sure packages can find python3-numpy libraries and
+# headers on the target architecture rather than the host, as well as making
+# sure the gfortran cross compiler is properly identified.
+
+# Even for cross compilation, numpy should be available on the host to ensure
+# that the host interpreter doesn't complain about missing deps
+if [[ $hostmakedepends != *"python3-numpy" ]]; then
+	hostmakedepends+=" python3-numpy"
+fi
+
+if [ "$CROSS_BUILD" ]; then
+	if [[ $makedepends != *"python3-numpy"* ]]; then
+		makedepends+=" python3-numpy"
+	fi
```

Comment:
Most of the other helpers seem to add stuff unconditionally. Why not here?

It could work like `add numpy helper -> numpy is added as a depend in the appropriate places, no need to worry about it`, or do you think people should still list `python3-numpy` somewhere in their dependencies? If so, what's the correct place?

I would have something like

```
makedepends+=" python3-numpy"

if [ "$CROSS_BUILD" ]; then
  hostmakedepends+=" python3-numpy"
[...]
```

Reply:
> Most of the other helpers seem to add stuff unconditionally. Why not here?

This isn't true. The `gir.sh` and `qemu.sh` helpers both do conditional appends to `hostmakedepends` and `makedepends` as is done here. The other helpers, `qmake.sh` and `rust.sh`, don't modify `*depends` at all.

> I would have something like [...]

Switching `makedepends` and `hostmakedepends` would have the same outcome, but I think it breaks with "correct" templating for Python. In many cases, Python packages have to be in `hostmakedepends` so that the host interpreter can run them or at least confirm that they are available at build time. (I actually think there are many templates that get this wrong; the fact that `setuptools` will use `pip` or `easy_install` to download dependencies on the host often hides this mistake by populating `wrksrc` with Python eggs to satisfy missing dependencies.) Often, Python packages (especially pure Python) never need to be in `makedepends` at all.

Packages like `python3-scipy` are special cases because they link C and FORTRAN against a library provided with `python3-numpy`. That needs to be in `makedepends` but, on native builds, we happen to already have provided the needed dependency.

To improve consistency with best practice for other Python dependencies, I think `hostmakedepends` should be the "standard" variable here, with `makedepends` added to fill the hole for cross builds.

Reply:
I see, that makes sense. Do you think we could add some of this info to the manual, as well? I think it would help a lot. Also with information about the helper and how to best use it - so always add numpy to `hostmakedepends`, while `makedepends` should only be touched by the helper, if I understand correctly.

Reply:
The helper adds the deps where they are necessary and not already provided by the template. I've been meaning to add a description of the new helper in the manual and will push that later today.

Maybe the host/target dependency issue for Python could use some specific attention in the manual, but that is more general and belongs in a separate PR. I'll give it some thought and put something together later.